python/hsfs/engine/spark.py

```diff
@@ -213,13 +220,16 @@ def register_hudi_temporary_table(
             self._spark_context,
             self._spark_session,
         )
+
         hudi_engine_instance.register_temporary_table(
             hudi_fg_alias,
             read_options,
         )
-        hudi_engine_instance.reconcile_hudi_schema(
-            self.save_empty_dataframe, hudi_fg_alias, read_options
-        )
+
+        if (hudi_fg_alias._feature_group.storage_connector is None):
```

Comment:
I think `reconcile_hudi_schema` is needed even if there is storage_connector

Reply:
we don't have the hive schema to compare against. (but i guess we could use FG features though)